### PR TITLE
[AMDGPU][SplitModule] Fix unintentional integer division

### PR DESCRIPTION
**Pull request by frasercrmck**
A static analysis tool warned that a division was always being performed
in integer division, so was either 0.0 or 1.0.

This doesn't seem intentional, so has been fixed to return a true ratio
using floating-point division. This in turn showed a bug where a
comparison against this ratio was incorrect.

### CODE REVIEW

File: llvm/test/tools/llvm-split/AMDGPU/large-kernels-merging.ll [hunk 1]
```diff
@@ -15,19 +15,20 @@
 ; Also check w/o large kernels processing to verify they are indeed handled
 ; differently.
 
-; P0 is empty
```
Comment (frasercrmck) [inline llvm/test/tools/llvm-split/AMDGPU/large-kernels-merging.ll:18]: I'm not claiming this is the correct fix for the test, but it gets it passing. I haven't updated the comments so those are now stale. @Pierre-vh what would you expect this test to check, with a fixed ratio?

**Comment by arsenm** [comment]
Description should be more specific

**Comment by frasercrmck** [comment]
> Description should be more specific

Yep, you're right, sorry. I've just split it into two PRs since we can't merge individual commits in the same PR

File: llvm/lib/Target/AMDGPU/AMDGPUSplitModule.cpp [hunk 2]
```diff
@@ -1101,7 +1101,7 @@ void RecursiveSearchSplitting::pickPartition(unsigned Depth, unsigned Idx,
         // Check if the amount of code in common makes it worth it.
         assert(SimilarDepsCost && Entry.CostExcludingGraphEntryPoints);
         const double Ratio =
-            SimilarDepsCost / Entry.CostExcludingGraphEntryPoints;
+            (double)SimilarDepsCost / Entry.CostExcludingGraphEntryPoints;
         assert(Ratio >= 0.0 && Ratio <= 1.0);
         if (LargeFnOverlapForMerge > Ratio) {
```
[the remark this answers is not in the document]
Comment (frasercrmck) [reply llvm/lib/Target/AMDGPU/AMDGPUSplitModule.cpp]: Yep good spot, thanks

[the remark this answers is not in the document]
Comment (frasercrmck) [reply llvm/lib/Target/AMDGPU/AMDGPUSplitModule.cpp]: I can't reply inline with your other comment for some reason, but it does indeed reverse the changes to the test.

**Review by Pierre-vh — APPROVED** [review]

**Pull request merged**